File: video_5_maxAdjSum.cpp
// #include<iostream>
// #include<bits/stdc++.h>
// using namespace std;

/******************************** MEMOIZATION ************************************************ */
// int solve(vector<int> &nums, int idx, vector<int> &dp){
//     if(idx == 0) return nums[idx];
//     if(idx < 0) return 0;
//     if(dp[idx] != -1) dp[idx];

//     int pick = nums[idx]+solve(nums, idx-2, dp);
//     int nPick = solve(nums, idx-1, dp);

//     return dp[idx] = max(pick, nPick);
// }


// int nonadjacentSum(vector<int> &nums){
//     int idx = nums.size()-1;
//     vector<int> dp(idx+1, -1);
//     return solve(nums, idx, dp);
// }

/******************** MAX SPACE OPTIMIZATION ****************************************************/
// int nonadjacentSum(vector<int> &nums){
//     int n = nums.size();
//     int prev = nums[0], prev2 = 0;

//     for(int i = 1; i<n; i++){
//         int take = nums[i];
//         if(i>1) take+=prev2;

//         int ntake = prev;
//         int curr = max(take, ntake);
//         prev2 = prev;
//         prev = curr;
//     }
//     return prev;
// }

/**************************************** MAIN ************************************************** */
// int main(){
//     int element, size;
//     vector<int> nums;

//     cout<<"enter size of the array:";
//     cin>>size;

//     cout<<"enter the elements in the array: ";
//     for(int i = 0; i<size; i++){
//         cin>>element;
//         nums.push_back(element);
//     }

//     cout<<"max adjacent sum = "<<nonadjacentSum(nums);
// }

File: video_5_maxAdjSum.java
import java.util.*;
class video_5_maxAdjSum {
//method 1: memoization-->

    // public static int solve(int ind, int arr[], int dp[], int n){
    //     if(ind>=n) return 0;

    //     if(dp[ind] != -1) return dp[ind];

    //     int Pick = arr[ind]+solve(ind+2, arr, dp, n);
    //     int nPick = solve(ind+1, arr, dp, n);

    //     return dp[ind] = Math.max(Pick, nPick); 

    // }
    
    // int findMaxSum(int arr[], int n) {
    //     // code here
    //     int dp[] = new int [n];
    //     for(int i = 0; i<n; i++)
    //         dp[i] = -1;
        
    //     return solve(0, arr, dp, n);
    // }

//method 2: tabulation
    // public static int findMaxSum(int arr[], int n) {
    //     // code here
    //     int dp[] = new int[n];
    //     dp[0] = arr[0];

    //     for(int i = 1; i<n; i++){
    //         int Pick = arr[i];
    //         if(i>1)
    //             Pick+=dp[i-2];
    //         int nPick = dp[i-1];

    //         dp[i] = Math.max(Pick, nPick);
    //     }
    //     return dp[n-1];
    // }

//method 3: max space optimization

    public static int findMaxSum(int arr[], int n) {
        // code here
        int prev = arr[0];
        int prev2 = 0;

        for(int i = 1; i<n; i++){
            int take = arr[i];
            if(i>1)
            take+=prev2;
            int nTake = prev;

            int curr = Math.max(take, nTake);
            prev2 = prev;
            prev = curr;
        }
        return prev;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];
        
        for(int i = 0; i<n; i++)
            arr[i] = sc.nextInt();
        
        System.out.print(findMaxSum(arr, n));
    }
}